optaplanner-website-root/content/blog/2023/04/24/OptaPlanner-9-released.adoc

```diff
@@ -0,0 +1,97 @@
+= OptaPlanner 9 has been released
+rsynek
+2023-04-24
+:page-interpolate: true
+:jbake-type: post
+:jbake-tags: migration, jakarta, javax, javaee, quarkus, spring boot
+:jbake-social_media_share_image: migration-flowchart.png
+
+In the previous article link:../../02/21/OptaPlanner-9-is-coming.adoc[OptaPlanner 9 is coming] we explained why we are doing OptaPlanner 9.
+Now OptaPlanner 9.37.0.Final has been released to enable users migrate to Jakarta EE, Quarkus 3 and Spring Boot 3.
+
+== Differences between OptaPlanner 8 and 9
+
+OptaPlanner 8 and 9 are fully compatible regarding their APIs and equal feature-wise.
+The differences are coming from package-level changes brought by Jakarta EE 10 that impact the entire Java ecosystem.
+OptaPlanner has to react, too.
+We decided to make the transition smooth by releasing two parallel streams: 8 and 9.
+
+The following table summarizes the compatibility of both OptaPlanner streams:
+
+[cols="1,1,1"]
+|===
+| ^|OptaPlanner 8 ^|OptaPlanner 9
+
+|Jakarta EE
+>|8 (via Quarkus)
+>|10 (via Quarkus)
+
+|Quarkus
+>|2
+>|3
+
+|Spring Boot
+>|2
+>|3
+|===
+
+Apart from those, OptaPlanner 9 currently does not support DRL score calculation when used together with Quarkus.
+If you use the DRL score calculation and Quarkus, either wait with migration to OptaPlanner 9 or consider link:https://www.optaplanner.org/download/upgradeRecipe/drl-to-constraint-streams-migration.html[moving to the Constraint Streams API].
```

Comment:
```suggestion
If you use the DRL score calculation, consider link:https://www.optaplanner.org/download/upgradeRecipe/drl-to-constraint-streams-migration.html[moving to the Constraint Streams API] as DRL has long since been deprecated.
```